.github/workflows/image-publish.yml

```diff
@@ -64,10 +71,35 @@ jobs:
       - name: Docker Build and Push
         run: |
           SHORT_SHA1=$(git rev-parse --short=7 HEAD)
-          docker buildx build --platform linux/amd64 -f build/dockerfiles/assembly.Dockerfile --push -t quay.io/che-incubator/che-code:insiders -t quay.io/che-incubator/che-code:next -t quay.io/che-incubator/che-code:insiders-${SHORT_SHA1} .
+          if [[ ${{ env.BRANCH_NAME }} == main ]]; then
+            docker buildx build --platform linux/amd64 -f build/dockerfiles/assembly.Dockerfile --push -t quay.io/che-incubator/che-code:insiders -t quay.io/che-incubator/che-code:next -t quay.io/che-incubator/che-code:insiders-${SHORT_SHA1} .
+          elif [[ ${{ env.BRANCH_NAME }} =~ ^7\.[0-9]+\.[0-9]+$ ]]; then
+            docker buildx build --platform linux/amd64 -f build/dockerfiles/assembly.Dockerfile --push -t quay.io/che-incubator/che-code:${{env.BRANCH_NAME}}  . 
```

Comment:
I guess technically this is either a branch or a tag, but that's fine as Git often uses "branch" for a "branch-or-tag-or-other-identifier". +1.